--- jde/lisp/jde-import.el	(revision 77)
+++ jde/lisp/jde-import.el	(working copy)
@@ -495,15 +495,17 @@
                    (classname (jde-import-get-classname name))
                    (case-fold-search nil)
                    (number-of-matches 
-                    (substring (count-matches 
-                                (concat "\\b" classname "\\b")) 0 2)))
+                    (count-matches (concat "\\b" classname "\\b"))))
+              (when (stringp number-of-matches)
+                (setq number-of-matches
+                      (string-to-number (substring number-of-matches 0 2))))
               (if (or 
                    ;; If name is already listed in the set
                    ;; of required imports...
                    (member name required-imports)
                    ;;or the class is not reference in the file
                    ;;and is not an import of the whole package i.e. .*
-                   (and (< (string-to-number number-of-matches) 2)
+                   (and (< number-of-matches 2)
                         (not (string= classname "*")))
                    ;; or imports a class in the current package...
                    (and
