--- remem-orig/savutil/gbuf.c	2007-04-23 08:22:14.000000000 +0200
+++ remem/savutil/gbuf.c	2009-06-09 19:56:07.000000000 +0200
@@ -38,16 +38,6 @@
 #include "gbuf.h"
 #include "savant.h"
 
-/* Return the minimum of n and strlen(n).  This won't read beyond n characters of string s,
-   so if s isn't null terminated it won't matter so long as it's malloc'ed up to n. */
-size_t strnlen (char *s, size_t n)
-{
-  int i;
-  if (s == NULL) return(0);
-  for (i=0; ((i<n) && (s[i] != '\0')); i++);
-  return(i);
-}
-
 /* Create a new Growbuffer, point gbuf to it. */
 void init_GBuffer(GBuffer *gbuf)
 {
