--- whine.el~	2007-09-16 04:31:58.000000000 +0200
+++ whine.el	2007-09-16 04:32:18.000000000 +0200
@@ -236,13 +236,13 @@
                  (symbol-function 'whine-get-word)
                  "'"
                  )))
-  (whine "" (append whine-pain
+  (whine "\C-p" (append whine-pain
                '("Up, up, and away!"
                 "This sure is an exciting line!"
                 "Gee!  Let's do that again!"
                 "Beginning of buffer"
                 )))
-  (whine "" (append whine-pain
+  (whine "\C-n" (append whine-pain
                '("Don't make me go that way"
                 "I don't want to go there!"
                 "What an unpleasant line!"
@@ -250,45 +250,45 @@
                 "End of buffer"
                 "File not found"
                 )))
-  (whine "" (append whine-pain
+  (whine "\C-d" (append whine-pain
                       '("Now what was that character again?"
                         "NOW look what you've done!"
                         "Buffer is read-only")))
-  (whine "" '("Oh, no!  I've forgotten what I did!"))
-  (whine "u" '("Oh, no!  I've forgotten what I did!"))
-  (whine "" '("Please don't make me KILL again!"))
-  (whine "" (append whine-pain
+  (whine "\C-_" '("Oh, no!  I've forgotten what I did!"))
+  (whine "\C-xu" '("Oh, no!  I've forgotten what I did!"))
+  (whine "\C-k" '("Please don't make me KILL again!"))
+  (whine "\C-f" (append whine-pain
                       '("Forward ho!" "Oh, no!")))
-  (whine "" (append whine-pain
+  (whine "\C-b" (append whine-pain
                 '("Please, Mr. Human Person, don't make me go that way!"
                 "I am taken aback!"
                 "Backwards ha!"
                 "Catastrophic error -- world aborted."
                 "Career aborted -- unable to restart."
                 )))
-  (whine "" (append whine-pain '("kill! Kill! KILL!")))
-  (whine "b" (append whine-pain '("This buffer is horrible!"
+  (whine "\C-w" (append whine-pain '("kill! Kill! KILL!")))
+  (whine "\C-xb" (append whine-pain '("This buffer is horrible!"
                  "I don't want to be here!")))
-  (whine "" '("You redrew the screen, you goon!"
+  (whine "\C-l" '("You redrew the screen, you goon!"
                 "No, you can't have a clear message line."
                 "This is getting boring."
                 "Give up and go play Rogue."
                 "You just typed control-L to refresh the screen."
                 ))
 
-  (whine "%" '("Oh wow!  I couldn't have done it better myself!"))
+  (whine "\M-%" '("Oh wow!  I couldn't have done it better myself!"))
   (whine "\C-i" '("Let's get the hell out of here!"))
   (whine "\e>" (append whine-pain '("It's off to the ends of the earth!")))
   (whine "\e<" (append whine-pain
                        '("It's off to the beginnings of the earth!")))
-  (whine "$" insult-word)
+  (whine "\M-$" insult-word)
   (whine " " insult-word)
-  (whine "" whine-insults)
-  (whine "b" whine-insults)
-  (whine "f" whine-insults)
-  (whine "" whine-insults)
-  (whine "2" (append whine-pain '("That just cuts me up.")))
-  (whine "x" '("You shouldn't have typed that command!"
+  (whine "\C-a" whine-insults)
+  (whine "\M-b" whine-insults)
+  (whine "\M-f" whine-insults)
+  (whine "\C-e" whine-insults)
+  (whine "\C-x2" (append whine-pain '("That just cuts me up.")))
+  (whine "\M-x" '("You shouldn't have typed that command!"
                  "That's not a legal command."
                  ))
   (whine "\C-x\C-s" '("Filesystem not writable."
@@ -299,19 +299,17 @@
                       "File saved.  Hallelujah!"
                       "File accidentally deleted -- please retype."
                       ))
-  (whine "" (append whine-pain '("If you touch that key again I shall explode!"
+  (whine "\C-m" (append whine-pain '("If you touch that key again I shall explode!"
                 "If you touch that key again I shall freak out!"
                 "That's the space bar, not the return key!"
                 "Press `c-c c-#' to abort."
                 "I won't RETURN!"
                 "You shouldn't have done that..."
                 "Now look what you've done!")))
-  (whine "" whine-pain)
-  (whine "v" whine-pain)
+  (whine "\C-v" whine-pain)
+  (whine "\M-v" whine-pain)
   ))
                 
-(whinify)
-
 (defun unwhinify ()
   "Hopefully, remove the whine-bindings on the whiny keys."
   (interactive)
