--- lib-src/yow.c~	1999-02-26 12:37:05.000000000 +0100
+++ lib-src/yow.c	2007-11-18 13:35:33.000000000 +0100
@@ -12,7 +12,8 @@
 
 #include <stdio.h>
 #include <ctype.h>
-#include <../src/epaths.h>      /* For PATH_DATA.  */
+#include <stdlib.h>
+#include <string.h>
 
 #define BUFSIZE  80
 #define SEP      '\0'
@@ -33,8 +34,6 @@
     &res;})
 #endif
 
-char *malloc(), *realloc();
-
 void yow();
 void setup_yow();
 
@@ -161,5 +160,6 @@
   }
   buf[i++] = 0;
   printf("%s\n", buf);
+  free (buf);
 }
 
