diff -r 66bbea0adac4 Makefile.in.in
--- a/Makefile.in.in	Mon Apr 07 08:55:10 2014 +0200
+++ b/Makefile.in.in	Tue Apr 08 00:02:42 2014 +0200
@@ -408,7 +408,7 @@
 	   fi; \
 	   ${INSTALL_DATA} lib-src/config.values $(DESTDIR)${docdir}/config.values; \
 	   ${INSTALL_DATA} lib-src/DOC $(DESTDIR)${docdir}/DOC; \
-	   for subdir in `find ${archlibdir} -type d ! -name RCS ! -name SCCS ! -name CVS -print` ; \
+	   for subdir in `find $(DESTDIR)${archlibdir} -type d ! -name RCS ! -name SCCS ! -name CVS -print` ; \
 	     do (cd $${subdir} && $(RM) -r RCS CVS SCCS \#* *~) ; done ; \
 	else true; fi
 #if (defined(PDUMP) && !defined (DUMP_IN_EXEC)) || (defined (PDUMP) && defined(WIN32_NATIVE))

