--- rudel-0.2-4-orig/rudel-icons.el
+++ rudel-0.2-4/rudel-icons.el
@@ -41,15 +41,17 @@
 ;;; Image constants
 ;;
 
+(eval-and-compile
 (defconst rudel-icons-image-formats '(svg png)
   "Image formats to try (in that order) when loading Rudel icons.")
 
-(defconst rudel-icons-directory
+(defvar rudel-icons-directory
   (file-name-as-directory
    (concat (file-name-directory
 	    (locate-library "rudel-icons.el"))
 	   "icons"))
   "Directory that holds Rudel icon files.")
+)
 
 
 ;;; Helper macro
