--- mairix.el~	2008-07-14 22:10:34.000000000 +0200
+++ mairix.el	2008-07-27 12:56:44.000000000 +0200
@@ -26,7 +26,8 @@
 ;; Features of mairix.el:
 ;;
 ;; * Query mairix with a search term.
-;; * Currently supported Emacs mail programs: RMail and Gnus (mbox only).
+;; * Currently supported Emacs mail programs: RMail, Gnus (mbox only),
+;;   and VM.
 ;; * Generate search queries using graphical widgets.
 ;; * Generate search queries based on currently displayed mail.
 ;; * Save regularly used searches in your .emacs customize section.
@@ -42,10 +43,10 @@
 ;; has more features and is better integrated with Gnus.  This
 ;; interface is essentially a stripped down version of nnmairix.el.
 ;;
-;; Currently, RMail and Gnus (with mbox files) are supported as mail
-;; programs, but it is pretty easy to interface it with other ones as
-;; well.  Please see the docs and the source for details.  In a
-;; nutshell: include your favourite mail program in
+;; Currently, RMail, Gnus (with mbox files), and VM are supported as
+;; mail programs, but it is pretty easy to interface it with other
+;; ones as well.  Please see the docs and the source for details.
+;; In a nutshell: include your favourite mail program in
 ;; `mairix-mail-program' and write functions for
 ;; `mairix-display-functions' and `mairix-get-mail-header-functions'.
 ;; If you have written such functions for your Emacs mail program of
@@ -142,15 +143,17 @@
 
 (defcustom mairix-mail-program 'rmail
   "Mail program used to display search results.
-Currently RMail and Gnus (mbox) are supported.  If you use Gnus
+Currently RMail, Gnus (mbox), and VM are supported.  If you use Gnus
 with maildir, use nnmairix.el instead."
   :type '(choice (const :tag "RMail" rmail)
-		 (const :tag "Gnus mbox" gnus))
+		 (const :tag "Gnus mbox" gnus)
+		 (const :tag "VM" vm))
   :group 'mairix)
 
 (defcustom mairix-display-functions
   '((rmail mairix-rmail-display)
-    (gnus mairix-gnus-ephemeral-nndoc))
+    (gnus mairix-gnus-ephemeral-nndoc)
+    (vm mairix-vm-display))
   "Specifies which function should be called for displaying search results.
 This is an alist where each entry consists of a symbol from
 `mairix-mail-program' and the corresponding function for
@@ -162,7 +165,8 @@
 
 (defcustom mairix-get-mail-header-functions
   '((rmail mairix-rmail-fetch-field)
-    (gnus mairix-gnus-fetch-field))
+    (gnus mairix-gnus-fetch-field)
+    (vm mairix-vm-fetch-field))
   "Specifies function for obtaining a header field from the current mail.
 This is an alist where each entry consists of a symbol from
 `mairix-mail-program' and the corresponding function for
@@ -286,6 +290,39 @@
     (gnus-summary-toggle-header 1)
     (message-field-value field)))
 
+;;; VM
+
+(eval-when-compile
+  (require 'vm nil t)
+  (require 'vm-folder nil t)
+  (require 'vm-macro nil t)
+  (require 'vm-misc nil t)
+  (require 'vm-summary nil t))
+
+;; Display function
+(defun mairix-vm-display (folder)
+  "Display mbox file FOLDER with VM."
+  (require 'vm)
+  ;; check if folder is already open and if so, kill it
+  (let ((buf (get-file-buffer folder)))
+    (when buf
+      (set-buffer buf)
+      (set-buffer-modified-p nil)
+      (condition-case nil
+	  (vm-quit t)
+	(error nil))
+      (kill-buffer buf)))
+  (vm-visit-folder folder t))
+
+;; Fetching mail header field
+(defun mairix-vm-fetch-field (field)
+  "Get mail header FIELD for current message using VM."
+  (save-excursion
+    (vm-select-folder-buffer)
+    (vm-check-for-killed-summary)
+    (vm-error-if-folder-empty)
+    (vm-get-header-contents
+     (car (vm-select-marked-or-prefixed-messages 1)) field)))
 
 ;;;; Main interactive functions
 
--- mairix-el.texi~	2008-07-27 07:37:07.000000000 +0200
+++ mairix-el.texi	2008-07-27 12:54:16.000000000 +0200
@@ -83,9 +83,10 @@
 widgets, similar to a customization buffer.
 
 Currently, @code{mairix.el} is only tested with mbox output together
-with RMail or Gnus as the Emacs mail program.  However, it should also
-work with Maildir or MH, and it should be very easy to integrate other
-Emacs mail programs into @code{mairix.el} (@pxref{Extending mairix.el}).
+with RMail, Gnus, or VM as the Emacs mail program.  However, it should
+also work with Maildir or MH, and it should be very easy to integrate
+other Emacs mail programs into @code{mairix.el}
+(@pxref{Extending mairix.el}).
 
 If you use Gnus with maildir or MH, you should really use the native
 Gnus back end @code{nnmairix} instead, since it is more tightly
@@ -154,9 +155,9 @@
 mairix.el.  The most important items are @emph{Mairix File Path},
 @emph{Mairix Search File} and @emph{Mairix Mail Program}.  The latter
 specifies which mail program should be used to display the mairix search
-results.  Currently, RMail and Gnus with mbox files are supported.  If
-you use Gnus with maildir or mh, use the native Gnus back end nnmairix
-instead.
+results.  Currently, RMail, Gnus with mbox files, and VM are supported.
+If you use Gnus with maildir or mh, use the native Gnus back end
+nnmairix instead.
 
 If you use another Emacs mail program which is not yet supported by
 mairix.el, it is pretty easy to integrate it.  @xref{Extending
