--- ermacs-20070104-orig/lib/slang/c_src/slang_drv.c	2003-02-21 20:17:57.000000000 +0100
+++ ermacs-20070104/lib/slang/c_src/slang_drv.c	2007-07-16 21:07:21.000000000 +0200
@@ -1,4 +1,5 @@
-
+#include <stdlib.h>
+#include <string.h>
 #include <stdio.h>
 #include "driver.h"
 #include <slang.h>
@@ -598,7 +599,7 @@
     case SMG_READ_RAW: {
 	x = get_int32(buf); buf+= 4;
 	t1 = malloc((2*x) + 2 + 1);
-	y = SLsmg_read_raw((unsigned short*)t1 +1, x);
+	y = SLsmg_read_raw((SLsmg_Char_Type *)t1 +1, x);
 	t1[1] = 1;
 	driver_output(port, t1, y+1);
 	free(t1);
