--- /local/home/ulm/dvi-view.el.orig	2007-08-23 15:13:10.000000000 +0200
+++ /local/home/ulm/dvi-view.el	2007-08-23 15:12:49.000000000 +0200
@@ -114,7 +114,7 @@
 				dvi-view-image-options))
 		  (start (point)))
 	      (if two-page-spread
-		  (if (evenp (car page))
+		  (if (zerop (mod (car page) 2))
 		      (when dvi-view-even-page-header
 			(insert (format dvi-view-even-page-header (car page))))
 		    (when dvi-view-odd-page-header
@@ -123,7 +123,7 @@
 		  (insert (format dvi-view-page-header (car page)))))
 	      (insert-image image)
 	      (if two-page-spread
-		  (if (evenp (car page))
+		  (if (zerop (mod (car page) 2))
 		      (when dvi-view-even-page-footer
 			(insert (format dvi-view-even-page-footer (car page))))
 		    (when dvi-view-odd-page-footer
